TF-6685 Add custom permissions documentation to team project access docs (#371)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/changelog.mdx

Keep track of changes to the API for Terraform Cloud and Terraform Enterprise.

## 2023-07-26

> * Introduce support for a `custom` `team_project` access level and various customizable permissions. There are `project-access` permissions that apply to the project itself as well as `workspace-access` permissions that apply to all workspaces in the project. For more information see the documentation on [Project Team Access](/terraform/cloud-docs/api-docs/project-team-access).

Reviewer: Suggestion for brevity:
```suggestion
* Add support for a `custom` option to the `team_project` access level along with various customizable permissions. The `project-access` permissions apply to the project itself, and `workspace-access` permissions apply to all workspaces within the project. For more information, see [Project Team Access](/terraform/cloud-docs/api-docs/project-team-access).
```
